core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -160,21 +167,16 @@ public String toString() {
     protected final MethodMetadata metadata;
     private final Map<Integer, Expander> indexToExpander = new LinkedHashMap<Integer, Expander>();
 
-    private BuildTemplateByResolvingArgs(MethodMetadata metadata) {
+    private BuildTemplateByResolvingArgs(MethodMetadata metadata, Factory paramExpanderFactory) {
       this.metadata = metadata;
+      checkNotNull(paramExpanderFactory, "paramExpanderFactory");
       if (metadata.indexToExpanderClass().isEmpty()) {
         return;
       }
       for (Entry<Integer, Class<? extends Expander>> indexToExpanderClass : metadata
```

Comment:
Hi @adriancole ... Indeed I did misunderstand you in gitter. You suggest moving the instantiation logic to `Contract.Default`, but would it be better to add this to `BaseContract`? In any case, this still doesn't maintain compatibility for any `Contract` implementations, if any, that exist in the wild and do not inherit from `BaseContract`. For this reason it might still be a good idea to, someplace, maintain a backup of the original `#newInstance()` invocation against the old `indexToxpanderClass` mapping. Any thoughts on these ideas?